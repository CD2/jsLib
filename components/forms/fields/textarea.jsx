import React from 'react'
import PropTypes from 'prop-types'

export default class TextArea extends React.Component {

  static propTypes = {
    initialValue: PropTypes.string,
    name: PropTypes.string,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onKeyPress: PropTypes.func,
    onRawChange: PropTypes.func,
    placeholder: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.object,
  }

  handleChange = (e) => {
    const { onRawChange, onChange } = this.props
    if (onRawChange) onRawChange(e)
    if (onChange) onChange({ name: e.target.name, value: e.target.value })
  }

  render() {
    const { name, value, placeholder, initialValue, onFocus, onKeyPress } = this.props

    return (
      <textarea
        name={name}
        defaultValue={initialValue}
        value={value || ``}
        placeholder={placeholder}
        onChange={this.handleChange}
        onFocus={onFocus}
        onKeyPress={onKeyPress}
      />
    )
  }

}
