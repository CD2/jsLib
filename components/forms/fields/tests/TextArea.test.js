/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { TextArea } from '../textarea'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`<TextArea />`, () => {
  const view = shallow((
    <TextArea />
  ))

  it(`snapshot`, ()=>expect(toJson(view)).toMatchSnapshot())
})