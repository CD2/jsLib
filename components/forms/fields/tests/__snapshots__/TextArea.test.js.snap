// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> snapshot 1`] = `
<textarea
  onChange={[Function]}
  value=""
/>
`;

exports[`snapshot 1`] = `
<textarea
  onChange={[Function]}
  value=""
/>
`;
